### PR TITLE
Cd 8

### PR DESCRIPTION
**Pull request by mariemm97**
Add new driver RabbitMQ nodes. 
Add comments.

### CODE REVIEW

File: sandbox_emulator/.env copy [hunk 1]
```diff
@@ -0,0 +1,12 @@
+DEVICE_IP=18.102.29.37
```
Comment (mohamedkallel) [inline sandbox_emulator/.env copy]: this file should not be commited, it's name should be .env.sample.{device_name} to be ignored

File: examples/http/rabbitmq_cluster.js [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+ */
+function extractValue(data, key) {
+    return function () {
+        return data[key] || 0;
+    };
+}
+
+// The list of custom driver variables to monitor
+function fillConfig() {
+    monitoringList = [
```
Comment (mohamedkallel) [inline examples/http/rabbitmq_cluster.js:102]: add comment for each monitoring variable

File: examples/http/rabbitmq_cluster.js [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+        },
+        {
+            uid: "channels",
+            label: "Channels total",
+            execute: extractValue(overview.object_totals, "channels")
+        }
+    ];
+}
+
+function fillTable() {
```
Comment (mohamedkallel) [inline examples/http/rabbitmq_cluster.js:239]: add comment what the table will contain

File: examples/http/rabbitmq_cluster.js [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+            return D.device.createVariable(c.uid, c.label, result, c.unit, c.type);
+        } else {
+            return null;
+        }
+    }).filter(function (v) {
+        return v != null;
+    });
+}
+
+//Load all data
```
Comment (mohamedkallel) [inline examples/http/rabbitmq_cluster.js:302]: explain more which data will be loaded

File: examples/http/rabbitmq_nodes.js [hunk 5]
```diff
@@ -0,0 +1,317 @@
+/** 
+ * This driver gets node metrics from RabbitMQ management plugin provided an HTTP-based API using HTTP agent
+ * Communication protocol is http
+ * Communicate with http api using USERNAME and PASSWORD
+ * The rabbitmq cluster name is modified in each installation.So, you can find it in the right side of the rabbitmq GUI. example: Cluster rabbit@319ac541e311
```
Comment (mohamedkallel) [inline examples/http/rabbitmq_nodes.js:5]: Correct it by:
The rabbitmq cluster name is modified in each installation. it should be found in in the top right corner of the RabbitMQ web application after making the signin. example: Cluster rabbit@319ac541e311

File: examples/http/rabbitmq_nodes.js [hunk 6]
```diff
@@ -0,0 +1,317 @@
...
+            return D.device.createVariable(c.uid, c.label, result, c.unit, c.type);
+        } else {
+            return null;
+        }
+    }).filter(function (v) {
+        return v != null;
+    });
+}
+
+// load all data
```
Comment (mohamedkallel) [inline examples/http/rabbitmq_nodes.js:274]: explain more which what data is loaded

File: examples/http/rabbitmq_nodes.js [hunk 7]
```diff
@@ -0,0 +1,317 @@
...
+function extractValue(data, key) {
+    return function () {
+        return data[key];
+    };
+}
+
+// The list of custom driver variables to monitor
+function fillConfig() {
+    monitoringList = [
+        {
```
Comment (mohamedkallel) [inline examples/http/rabbitmq_nodes.js:144]: add a description for each variable

**Review by mohamedkallel — CHANGES_REQUESTED** [review]
the code is good, just fix the comments I added

**Review by GeorgiTsatsev — APPROVED** [review]

**Review by GeorgiTsatsev — APPROVED** [review]

**Review by GeorgiTsatsev — APPROVED** [review]

**Review by mohamedkallel — APPROVED** [review]

**Pull request merged**